cpp/src/groupby/sort/groupby.cu

```diff
@@ -0,0 +1,348 @@
+    } else {
+      return compute_sort_groupby<true, false>;
+    }
+  } else {
+    if (cudf::has_nulls(values)) {
+      return compute_sort_groupby<false, true>;
+    } else {
+      return compute_sort_groupby<false, false>;
+    }
+  }
+}
+} // anonymous namespace
+
+namespace detail { 
+ 
+
+/**---------------------------------------------------------------------------*
+ * @brief Verifies the requested aggregation is valid for the arguments of the 
+ * operator.
+ *
+ * @throw cudf::logic_error if an invalid combination of argument and operator
+ * is requested.
+ *
+ * @param ops The aggregation operators
+ * @param ops The aggregation arguments
+ *---------------------------------------------------------------------------**/
+static void verify_operators_with_arguments(std::vector<operators> const& ops, std::vector<operation_args*> const& args) {
+   CUDF_EXPECTS(ops.size() == args.size(),
+               "Size mismatch between ops and args");
+  for (size_t i = 0; i < ops.size(); i++) {
+    if(ops[i] == QUANTILE) { 
+      quantile_args* q_args = static_cast<quantile_args*>(args[i]); 
+      if (q_args == nullptr or q_args->quantiles.size() == 0) {
+        CUDF_FAIL(
+                "Cannot compute QUANTILE aggregation. It requires quantiles argument.");
+      }
+    } 
+  }
+}
+
+std::pair<cudf::table, std::vector<gdf_column*>> groupby(cudf::table const& keys,
+                                            cudf::table const& values,
+                                            std::vector<operation> const& ops,
+                                            Options options,
+                                            cudaStream_t stream) {
+  CUDF_EXPECTS(keys.num_rows() == values.num_rows(),
+               "Size mismatch between number of rows in keys and values.");
+  std::vector<operators> optype_list(ops.size());
+  std::transform(ops.begin(), ops.end(), optype_list.begin(), [](auto const& op) {
+    return op.op_name;
+  });
+  std::vector<operation_args*> ops_args(ops.size());
+  std::transform(ops.begin(), ops.end(), ops_args.begin(), [](auto const& op) {
+    return op.args.get();
+  });
+  verify_operators(values, optype_list);
+  verify_operators_with_arguments(optype_list, ops_args);
+
+  // Empty inputs
+  if (keys.num_rows() == 0) {
+    cudf::table output_values(0, target_dtypes(column_dtypes(values), optype_list), column_dtype_infos(values));
+    return std::make_pair(
+        cudf::empty_like(keys),
+        std::vector<gdf_column*>{output_values.begin(), output_values.end()}
+        );
+  }
+
+  auto compute_groupby = groupby_null_specialization(keys, values);
+
+  cudf::table output_keys;
+  std::vector<gdf_column*> output_values;
+  std::tie(output_keys, output_values) =
+      compute_groupby(keys, values, optype_list, ops_args, options, stream);
+  
+  cudf::table table_output_values(output_values);
+  
+  update_nvcategories(keys, output_keys, values, table_output_values);
```

Comment:
Check return value and throw exception if it fails.

Reply:
Unresolving this because it's not OK that update_nvcategories can fail silently. Please modify it to check for error return values from the functions it calls.